common/functions/src/scalars/strings/locate.rs

```diff
@@ -22,14 +22,54 @@ use crate::scalars::function_factory::FunctionDescription;
 use crate::scalars::function_factory::FunctionFeatures;
 use crate::scalars::Function;
 
+const FUNC_LOCATE: u8 = 1;
+const FUNC_POSITION: u8 = 2;
+const FUNC_INSTR: u8 = 3;
+
+pub struct LocateFunction {}
+
+impl LocateFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        LocatingFunction::<FUNC_LOCATE>::try_create(display_name)
+    }
+
+    pub fn desc() -> FunctionDescription {
+        LocatingFunction::<FUNC_LOCATE>::desc()
+    }
+}
+
+pub struct InstrFunction {}
+
+impl InstrFunction {
```

Comment:
type LocateFunction =  LocatingFunction::<FUNC_LOCATE>;
type InstrFunction  = LocatingFunction::<FUNC_INSTR>;

Reply:
thank you, changed

Reply:
Besides, `String2String` function is a good template function to learn.